gradle-failure-reports/src/main/java/com/palantir/gradle/failurereports/LocalFailuresRecommendationsPlugin.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.failurereports;
+
+import com.palantir.gradle.failurereports.util.PluginResources;
+import javax.inject.Inject;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.flow.FlowProviders;
+import org.gradle.api.flow.FlowScope;
+
+public abstract class LocalFailuresRecommendationsPlugin implements Plugin<Project> {
+
+    @Inject
+    protected abstract FlowScope getFlowScope();
```

Comment:
TIL you can inject not only this but a bunch of other Build/Global scoped things directly into the plugin!